stan/math/gpu/copy_triangular_opencl.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef STAN_MATH_GPU_COPY_TRIANGULAR_OPENCL_HPP
+#define STAN_MATH_GPU_COPY_TRIANGULAR_OPENCL_HPP
+#ifdef STAN_OPENCL
+#include <stan/math/gpu/matrix_gpu.hpp>
+#include <CL/cl.hpp>
+
+namespace stan {
+namespace math {
+enum triangularity { LOWER = 0, UPPER = 1, NONE = 2 };
+
+/**
+ * Copies the lower or upper
+ * triangular of the source matrix to
+ * the destination matrix.
+ * Both matrices are stored on the GPU.
+ *
+ * @param src the source matrix
+ * @param lower_upper enum to describe
+ * which part of the matrix to copy:
+ * LOWER - copies the lower triangular
+ * UPPER - copes the upper triangular
+ *
+ * @return the matrix with the copied content
+ *
+ */
+inline matrix_gpu copy_triangular(matrix_gpu& src, triangularity lower_upper) {
+  if (src.size() == 0) {
+    return src;
+  }
+  if (src.size() == 1) {
+    return src;
```

Comment:
Are there any memory aliasing issues we need to worry about? (My guess is no and this creates a copy, but would like verification)

Reply:
In case of sizes 0 and 1 this does not create a copy. I will changes this so that it does.